framework/src/executioners/PicardSolve.C

```diff
@@ -136,44 +141,56 @@ PicardSolve::solve()
     {
       if (_has_picard_its)
       {
-        _console << "\n 0 Picard |R| = "
-                 << Console::outputNorm(std::numeric_limits<Real>::max(), _picard_initial_norm)
-                 << '\n';
-
-        for (unsigned int i = 0; i <= _picard_it; ++i)
+        if (_has_picard_norm)
```

Comment:
What I was ideally hoping was to have a way to do at most `picard_max_its` iterations before going to the next time step, i.e. go to the next time step if either the tolerances are met or the maximum number of Picard iterations is reached. For pseudo-transients, it makes sense to impose a fixed number of iterations for the 'early' time steps but not when we have almost reached convergence. Here it looks like the tolerances are completely disregarded if _has_picard_norm is true. Maybe you have specific applications in mind?

Reply:
The parameter I added here is serving duel purpose: 1. avoid residual norm evaluations; 2. not fail the picard solve even maximum of picard iteration is reached. Looks like you want to have two separate parameter for the two goals. Yes, you are right, turning on the current parameter in this PR will disable the residual norm checks completely although it does allow user custom convergence checks.

Reply:
When you say pseudo-transients, are both the master and subapp pseudo-transient? I guess at least mater is pseudo, right?

Reply:
Yes, both are